src/Kdyby/Doctrine/EntityRepository.php

```diff
@@ -74,11 +74,13 @@ public function findOneBy(array $criteria, array $orderBy = null)
 
 	public function countBy(array $criteria = array())
 	{
-		return (int) $this->createQueryBuilder('e')
+		$result = $this->createQueryBuilder('e')
 			->whereCriteria($criteria)
 			->select('COUNT(e)')
 			->setMaxResults(1)
-			->getQuery()->getSingleScalarResult();
+			->getQuery()->getSingleResult();
+
+		return (int) $result[1];
```

Comment:
Works MySql + MSSql :+1: 

btw: query itself is same :) SELECT COUNT(m0_.id) AS sclr_0 FROM mail_history m0_ LIMIT 1, Doctrine parses  `rowsCount` inside 

dump is `array (1): rowsCount => "20" (2)`